substratevm/src/com.oracle.svm.native.jvm.posix/src/JvmFuncs.c

```diff
@@ -78,7 +88,24 @@ JNIEXPORT int JNICALL JVM_Connect(int fd, struct sockaddr* him, socklen_t len) {
 }
 
 JNIEXPORT void* JNICALL JVM_FindLibraryEntry(void* handle, const char* name) {
-    return dlsym(handle, name);
+    /*
+        Calls to this function from a static binary are inherently unsafe. On some libc implementations, it may result
+        in a segfault, while on others, the symbol could be wrongly not found. As of JDK11, this function is invoked in
+        only one place: IPv6 support checking.
+        As a safeguard, we restrict access to only known used symbols from the JDK code. If a future version introduces
+        a dependency on another symbol, it could result in hard-to-find bugs. Therefor, calling this function from a
+        static binary with an unknown symbol terminates the program.
+    */
+    if (__svm_vm_is_static_binary) {
+        if (strcmp(name, "inet_pton") == 0) {
+            return inet_pton;
+        }
+        fprintf(stderr, "Internal error: JVM_FindLibraryEntry called from a static native image. Results may be unpredictable. Please report this issue to the SubstrateVM team.");
```

Comment:
Use
```
(*env)->FatalError(env, "Internal error: JVM_FindLibraryEntry called from a static native image. Results may be unpredictable. Please report this issue to the SubstrateVM team.");
```
instead. It has the advantage that we see the whole stack trace that caused the situation.

Reply:
Ahh that's not possible here because we don't get `JNIEnv` passed into `JVM_FindLibraryEntry`.